erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -754,13 +755,14 @@ def make_loyalty_point_redemption_gle(self, gl_entries):
 					"against": "Expense account - " + cstr(self.loyalty_redemption_account) + " for the Loyalty Program",
 					"credit": self.loyalty_amount,
 					"against_voucher": self.return_against if cint(self.is_return) else self.name,
-					"against_voucher_type": self.doctype
+					"against_voucher_type": self.doctype,
+					"cost_center": self.cost_center
 				})
 			)
 			gl_entries.append(
 				self.get_gl_dict({
 					"account": self.loyalty_redemption_account,
-					"cost_center": self.loyalty_redemption_cost_center,
+					"cost_center": self.cost_center or self.loyalty_redemption_cost_center,
```

Comment:
priority should be given to `loyalty_redemption_cost_center`

Reply:
If user opted to tag cost center in Balance Sheet Accounts then loyalty_redemption amount should also go to Cost Center selected in parent document otherwise trial balance wouldn't match.

Reply:
ok